src/test/reporters/base.ts

```diff
@@ -277,6 +272,38 @@ function indent(lines: string, tab: string) {
   return lines.replace(/^(?=.+$)/gm, tab);
 }
 
+export function prepareErrorStack(
+  stack: string,
+  file?: string
+): {
+  message: string;
+  stackLines: string[];
+  position?: PositionInFile;
+  codeFrame?: string;
+} {
+  const lines = stack.split('\n');
+  let firstStackLine = lines.findIndex(line => line.startsWith('    at '));
+  if (firstStackLine === -1) firstStackLine = lines.length;
+  const message = lines.slice(0, firstStackLine).join('\n');
+  const stackLines = lines.slice(firstStackLine);
+  const position = file ? positionInFile(stackLines, file) : undefined;
+  let codeFrame;
+  if (position) {
+    const source = fs.readFileSync(file!, 'utf8');
+    codeFrame = codeFrameColumns(
```

Comment:
I wouldn't do `codeFrame` in this method - it is some expensive unnecessary work for the json reporter. Why don't we leave it in `formatError`?

Reply:
Done